### ISSUE TITLE
Problem Updating to 2.2

### ISSUE THREAD

**Issue by cdw9** [collective/collective.easyslideshow#18]
While updating a Plone site from 4.1.3 > 4.3.4 and EasySlideshow 2.0 > 2.2, I got the following error when running buildout. It references `collective.easyslideshow.descriptors.SlideshowDescriptor`, which is removed in 2.2

```
Traceback (most recent call last):
  File "/usr/home/chrissy/projects/project/project-buildout/parts/instance/bin/interpreter", line 278, in <module>
    exec(_val)
  File "<string>", line 1, in <module>
  File "/usr/home/chrissy/.buildout/eggs/collective.recipe.plonesite-1.9.0-py2.7.egg/collective/recipe/plonesite/plonesite.py", line 332, in <module>
    main(app, parser)  # NOQA
  File "/usr/home/chrissy/.buildout/eggs/collective.recipe.plonesite-1.9.0-py2.7.egg/collective/recipe/plonesite/plonesite.py", line 238, in main
    traverse(vhm_string)
  File "/usr/home/chrissy/.buildout/eggs/Zope2-2.13.22-py2.7.egg/ZPublisher/BaseRequest.py", line 502, in traverse
    subobject = self.traverseName(object, entry_name)
  File "/usr/home/chrissy/.buildout/eggs/Zope2-2.13.22-py2.7.egg/ZPublisher/BaseRequest.py", line 340, in traverseName
    adapter = queryMultiAdapter((ob, self), IPublishTraverse)
  File "/usr/home/chrissy/.buildout/eggs/zope.component-3.9.5-py2.7.egg/zope/component/_api.py", line 120, in queryMultiAdapter
    return sitemanager.queryMultiAdapter(objects, interface, name, default)
  File "/usr/home/chrissy/.buildout/eggs/ZODB3-3.10.5-py2.7-freebsd-8.2-RELEASE-p3-i386.egg/ZODB/Connection.py", line 860, in setstate
    self._setstate(obj)
  File "/usr/home/chrissy/.buildout/eggs/ZODB3-3.10.5-py2.7-freebsd-8.2-RELEASE-p3-i386.egg/ZODB/Connection.py", line 914, in _setstate
    self._reader.setGhostState(obj, p)
  File "/usr/home/chrissy/.buildout/eggs/ZODB3-3.10.5-py2.7-freebsd-8.2-RELEASE-p3-i386.egg/ZODB/serialize.py", line 612, in setGhostState
    state = self.getState(pickle)
  File "/usr/home/chrissy/.buildout/eggs/ZODB3-3.10.5-py2.7-freebsd-8.2-RELEASE-p3-i386.egg/ZODB/serialize.py", line 605, in getState
    return unpickler.load()
  File "/usr/local/lib/python2.7/copy_reg.py", line 48, in _reconstructor
    obj = object.__new__(cls)
TypeError: ('object.__new__(SlideshowDescriptor) is not safe, use persistent.Persistent.__new__()', <function _reconstructor at 0x28528e9c>, (<class 'collective.easyslideshow.descriptors.SlideshowDescriptor'>, <type 'object'>, None))
```

**Comment by cdw9** [comment]
And also once Plone was upgraded to 4.3.4 with EasySlideshow 2.1, When I change the version to 2.2 and run buildout, there is still a similar error:

    2015-03-26 13:56:16 WARNING OFS.Uninstalled Could not import class 'SlideshowDescriptor' from module 'collective.easyslideshow.descriptors'
    2015-03-26 13:56:16 ERROR ZODB.Connection Couldn't load state for 0x24
    Traceback (most recent call last):
      File "/usr/home/chrissy/.buildout/eggs/ZODB3-3.10.5-py2.7-freebsd-8.2-RELEASE-p3-i386.egg/ZODB/Connection.py", line 860, in setstate
        self._setstate(obj)
      File "/usr/home/chrissy/.buildout/eggs/ZODB3-3.10.5-py2.7-freebsd-8.2-RELEASE-p3-i386.egg/ZODB/Connection.py", line 914, in _setstate
        self._reader.setGhostState(obj, p)
      File "/usr/home/chrissy/.buildout/eggs/ZODB3-3.10.5-py2.7-freebsd-8.2-RELEASE-p3-i386.egg/ZODB/serialize.py", line 612, in setGhostState
        state = self.getState(pickle)
      File "/usr/home/chrissy/.buildout/eggs/ZODB3-3.10.5-py2.7-freebsd-8.2-RELEASE-p3-i386.egg/ZODB/serialize.py", line 605, in getState
        return unpickler.load()
      File "/usr/local/lib/python2.7/copy_reg.py", line 48, in _reconstructor
        obj = object.__new__(cls)
    TypeError: ('object.__new__(SlideshowDescriptor) is not safe, use persistent.Persistent.__new__()', <function _reconstructor at 0x28528e9c>, (<class 'collective.easyslideshow.descriptors.SlideshowDescriptor'>, <type 'object'>, None))
    > /usr/local/lib/python2.7/copy_reg.py(48)_reconstructor()
    -> obj = object.__new__(cls)

**Comment by cdw9** [comment]
For anyone else that runs into this: wildcard.fixpersistentutilities was able to help.

Still need a fix in easyslideshow.

**Comment by cdw9** [comment]
This should now be fixed in version 2.3
